pkgs/development/libraries/libhugetlbfs/default.nix

```
@@ -32,6 +32,9 @@ stdenv.mkDerivation rec {
     "EXEDIR=$(bin)/bin"
     "DOCDIR=$(doc)/share/doc/libhugetlbfs"
     "MANDIR=$(man)/share/man"
+  ] ++ lib.optionals (stdenv.buildPlatform.system != stdenv.hostPlatform.system) [
+    # The ARCH logic defaults to querying `uname`, which will return build platform arch
+    "ARCH=${stdenv.hostPlatform.system}"
```

`uname -m` is `stdenv.hostPlatform.uname.processor`

```suggestion
    "ARCH=${stdenv.hostPlatform.uname.processor}"
```

---

thanks